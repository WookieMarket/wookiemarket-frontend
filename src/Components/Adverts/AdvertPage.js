import React from 'react';

const AdvertPage = ({ name, image, description, sale, price }) => {
  return (
    <div>
      <h2>{name}</h2>
      <img src={image} alt={name} />
      <p>{description}</p>
      <p>{sale ? 'For sale' : 'Purchase'}</p>
      <p>Price: {price}</p>
    </div>
  );
};

export default AdvertPage;
