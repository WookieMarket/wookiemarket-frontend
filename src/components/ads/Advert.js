import React from 'react';
import Fields from '../layout/Fields';
import defaultImage from '../../assets/no_image.jpg';
import '../../css/advert.css';

function Advert({ advert, onImageError }) {
  console.log('Rendering Advert component');
  //const advDate = new Date(advert.createdAt);
  if (!advert) {
    console.error('No advertisement with the specified ID found');
    return <p>The specified advertisement could not be found.</p>;
  }

  return (
    <>
      <div className='productInfo' id='advertOnly'>
        <div className='advert-name'>
          <h2>
            <Fields value={advert.name} />
          </h2>
        </div>
      </div>
      <br />
      <div className='productData'>
        <div className='product-img'>
          <img
            className='productPhoto'
            src={advert.image || defaultImage}
            onError={onImageError}
            alt='imagen del producto en venta'
          ></img>
        </div>
        <br />
        <div className='product_Info'>
          <p className='advert_label'>Description: </p>
          <Fields value={advert.description} />
          {/*<span className='description'> {advert.description}</span>*/}
          <br />
          <br />
          <p>
            Is
            {advert.onSale ? (
              <span id='isSale'> for sale </span>
            ) : (
              <span id='isSale'> purchased </span>
            )}
            this product by:
          </p>
          <div className='price'>
            <p>
              <Fields value={advert.price} />
              {/*{advert.price}*/}{' '}
              <span id='price_coin'>
                {' '}
                <Fields value={advert.coin} />
              </span>
            </p>
          </div>
          <div className='advert_label'>
            <p>Category: </p>
            <p>Category: </p>
            <span className='advert_text'>
              {advert.category &&
                advert.category.map((cat) => (
                  <Fields key={cat} value={cat.toLocaleUpperCase()} />
                ))}
            </span>
            {/*</span><span className='advert_text'>
                {advert.category === undefined
                  ? ''
                  : advert.category.join(', ').toLocaleUpperCase()}
              </span>
            */}
          </div>
        </div>
        <div>
          <p className='advert_label'>
            Username: <Fields value={advert.username} />
            {/*<span className='advert-text'>{advert.username}</span>*/}
          </p>
          <small className='advert_label'>
            Created at:{' '}
            {
              //<span className='advert_text'>{`${advDate.toUTCString()}`}</span>
            }
          </small>
        </div>
      </div>
    </>
  );
}

export default Advert;
