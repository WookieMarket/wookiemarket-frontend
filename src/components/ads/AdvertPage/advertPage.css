header,
footer {
  width: 100%;
}

.content {
  width: 100%;
  font-size: 2em;
}

.content h1 {
  margin-bottom: 5px;
  text-transform: lowercase;
  font-family: var(--font-family);
  font-size: 1.2rem;
}

.holoBase {
  z-index: 2;
  position: relative;
  bottom: -15px;
  width: 75%;
  height: var(--holo-height);
  margin: -5px auto 0;
  background-color: var(--background-navbar-color, 0.35);
  border-radius: 100%;
  filter: blur(2.5px);
  transform: perspective(100px) rotateX(-30deg);
  transform-style: preserve-3d;
  animation: holographic 1.5s infinite alternate;
  backface-visibility: hidden;

  &:after {
    position: absolute;
    left: -10px;
    right: -10px;
    content: '';
    height: 10px;
    background-color: var(--background-navbar-color, 0.5);
    border-radius: 100%;
    filter: blur(2.5px);
  }

  + .holo {
    animation-duration: 0.75s;
    opacity: 0.5;
  }
}

.productData {
  border: none;
}

.productInfo {
  margin-top: 10px;
}

@keyframes float {
  from {
    will-change: unset;
    transform: translateY(-10px);
  }

  to {
    will-change: transform;
    transform: translateY(-30px);
  }
}

@keyframes holographic {
  from {
    will-change: box-shadow;
    box-shadow:
      0 -75px 5px var(--holo-color, 0.15),
      0 -1px 2px var(--holo-color, 0.5),
      0 -45px 5px var(--holo-color, 0.15),
      0 -2px 3px var(--holo-color, 0.5),
      0 -3px 4px var(--holo-color, 0.5),
      0 -4px 6px var(--holo-color, 0.5),
      0 -17px 10px var(--holo-color, 0.75),
      0 -7px 20px var(--holo-color, 0.75),
      0 -10px 30px var(--holo-color, 0.75),
      0 -15px 40px var(--holo-color, 0.75),
      0 -25px 50px var(--holo-color, 0.75),
      0 -35px 60px var(--holo-color, 0.85),
      0 -45px 70px var(--holo-color, 0.95),
      0 -65px 80px var(--holo-color, 1),
      0 -75px 90px var(--holo-color, 1);
  }

  to {
    will-change: unset;
    box-shadow:
      0 -1px 5px var(--holo-color, 0.15),
      0 -1px 2px var(--holo-color, 0.5),
      0 -1px 5px var(--holo-color, 0.15),
      0 -2px 3px var(--holo-color, 0.5),
      0 -3px 4px var(--holo-color, 0.5),
      0 -4px 6px var(--holo-color, 0.5),
      0 -5px 7px var(--holo-color, 0.75),
      0 -7px 10px var(--holo-color, 0.75),
      0 -10px 15px var(--holo-color, 0.75),
      0 -15px 20px var(--holo-color, 0.75),
      0 -25px 25px var(--holo-color, 0.75),
      0 -35px 30px var(--holo-color, 0.85);
  }
}

.no-advert {
  background-image: var(--noAdvertFound-image);
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center center;
  border-radius: 25px;
  height: 75vh;
}

.holoBaseNoAdvert {
  display: none;
}

.StarWarsObject {
  display: inline-block;
  letter-spacing: 15px;
  font-family: var(--font-family-SraWarsObjects);
}

.fontInverted {
  -webkit-transform: scaleX(-1);
  -moz-transform: scaleX(-1);
  -ms-transform: scaleX(-1);
  -o-transform: scaleX(-1);
  transform: scaleX(-1);
}

.buttonSection {
  display: flex;
  justify-content: space-around;
  margin-bottom: 25px;
}

@media screen and (min-width: 768px) {
  .content h1 {
    font-size: 1.5rem;
  }
}

@media screen and (min-width: 1024px) {
  .content h1 {
    font-size: 2rem;
  }
}
