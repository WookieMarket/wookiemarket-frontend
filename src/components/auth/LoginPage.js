import { useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getUi } from "../../store/selectors";
import { resetError } from "../../store/slices/ui";
import { authLogin } from "../../store/slices/auth";
import ErrorModal from "../shared/modal/ErrorModal";
import Spinner from "../shared/spinner/Spinner";
import Layout from "../layout/Layout";

//DONE Log in with username and password and a checkbox to give the option to persist the token, also handle errors and user feedback. When doing Login I want to send the user to the page they wanted to go to.

function LoginPage() {
  const dispatch = useDispatch();
  const { isLoading, error } = useSelector(getUi);

  const [credentials, setCredentials] = useState({
    username: "",
    password: "",
    rememberMe: false,
  });

  const handleErrorClick = () => {
    dispatch(resetError());
  };

  const handleChange = event => {
    const { name, value, type, checked } = event.target;

    setCredentials(prevCredentials => ({
      ...prevCredentials,
      [name]: type === "checkbox" ? checked : value,
    }));
  };

  const handleSubmit = async event => {
    event.preventDefault();

    dispatch(authLogin(credentials));
  };

  const buttonDisabled =
    isLoading || !credentials.username || !credentials.password;

  return (
    <Layout title="Login Page">
      <div>
        {isLoading ? (
          <Spinner message="cargando..." />
        ) : (
          <form onSubmit={handleSubmit} className="container-form">
            <label className="form-label" htmlFor="username">
              Username
            </label>
            <input
              id="username"
              className="form-input"
              type="text"
              name="username"
              onChange={handleChange}
              value={credentials.username}
              required
            />
            <label className="form-label" htmlFor="password">
              Password
            </label>
            <input
              id="password"
              className="form-input"
              type="password"
              name="password"
              onChange={handleChange}
              value={credentials.password}
              required
            />
            <label htmlFor="rememberMe">
              rememberMe
              <input
                id="rememberMe"
                type="checkbox"
                name="rememberMe"
                checked={credentials.rememberMe}
                onChange={handleChange}
              />
            </label>
            <button
              data-testid="button"
              type="submit"
              //variant="primary"
              width="button-form"
              disabled={buttonDisabled}>
              Log in
            </button>
          </form>
        )}

        {error && (
          <ErrorModal
            title="Error"
            message={error.message}
            onCancel={handleErrorClick}
          />
        )}
      </div>
    </Layout>
  );
}

export default LoginPage;
