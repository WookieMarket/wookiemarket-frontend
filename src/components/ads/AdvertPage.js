import React from 'react';
import '../../css/advert.css';
import Advert from './Advert';
import { useLocation, useNavigate, useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { getAdvertById } from '../../store/selectors';

const AdvertPage = ({ onImageError }) => {
  const { advertId } = useParams();
  /*const location = useLocation();
  const searchParams = new URLSearchParams(location.search);
  const advertId = searchParams.get('_id');*/

  console.log('Advert ID:', advertId);

  const advert = useSelector((state) => getAdvertById(state, advertId));

  console.log('Esto debería ser el anuncio:', advert);
 

  const navigate = useNavigate();
  //navigate.replace(`/adverts/${advert.name}`);

  if (advert) {
    navigate(`/adverts/${advert._id}`);
  } else {
    console.error('No se encontró un anuncio con el ID especificado');
  }

  return (
    <>
      <Advert advert={advert} onImageError={onImageError} />
      <div>
        <small>Created at: ¡AÑADIR CAMPO FECHA CREACIÓN ANUNCIO!</small>
      </div>
    </>
  );
};

export default AdvertPage;
