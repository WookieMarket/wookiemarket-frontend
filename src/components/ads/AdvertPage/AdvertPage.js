import React from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';
import { getAdvertById } from '../../../store/selectors';
import Advert from '../Advert/Advert';

const AdvertPage = ({ advertId }) => {
  const advDate = new Date();
  const { t } = useTranslation();

  const advert = useSelector((state) => getAdvertById(state, advertId));
  return (
    <>
      <div className='content'>
        <Advert {...advert} />
      </div>
    </>
  );
};

export default AdvertPage;
