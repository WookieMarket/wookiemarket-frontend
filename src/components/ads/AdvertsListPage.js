import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import Advert from "./Advert";
import { useDispatch, useSelector } from "react-redux";
import { getAdverts, getUi } from "../../store/selectors";
import { advertsList } from "../../store/slices/ads";
import "../../css/advertListPage.css";
import "../../css/advert.css";
import defaultImage from "../../assets/no_image.jpg";
import { filterByName } from "../../store/slices/adsFiltered";
import Layout from "../layout/Layout";

const EmptyList = ({ dataFiltered }) => {
  return (
    <div style={{ textAlign: "center" }}>
      <p>Sorry, no adverts yet.</p>
    </div>
  );
};

const handleImageError = event => {
  event.target.src = defaultImage;
};

const advertsPerPage = 2;

const AdvertsListPage = () => {
  //const [query, setQuery] = useState('');
  const [currentPage, setCurrentPage] = useState(1);
  const [filteredAdverts, setFilteredAdverts] = useState("");
  const [isFilterActive, setIsFilterActive] = useState(false);
  const ads = useSelector(getAdverts);
  const { isLoading } = useSelector(getUi);
  const dispatch = useDispatch();

  // useEffect(() => {
  //   onAdvertsLoaded();
  // }, [onAdvertsLoaded]);

  useEffect(() => {
    dispatch(advertsList()).catch(error => console.log(error));
  }, [dispatch]);

  const handlePageChange = page => {
    setCurrentPage(page);
  };

  const filterAdName = ad =>
    (ad.name ?? "").toUpperCase().startsWith(filteredAdverts.toUpperCase());

  const filteredAds = ads
    //.filter(filterAdSaleValueOrDefault)
    .filter(filterAdName);
  //.filter(filterAdPrice)
  //.filter(filterAdTags);

  //const totalPages = Math.ceil(adverts.length / advertsPerPage);
  const totalPages = Math.ceil(
    (isFilterActive ? filteredAds.length : ads.length) / advertsPerPage,
  );
  const startIndex = (currentPage - 1) * advertsPerPage;
  const endIndex = startIndex + advertsPerPage;
  //const advertsToDisplay = adverts.slice(startIndex, endIndex);
  const advertsToDisplay = isFilterActive
    ? filteredAds.slice(startIndex, endIndex)
    : ads.slice(startIndex, endIndex);
  //const isLastPage = currentPage === totalPages;
  const isLastPage = isFilterActive
    ? currentPage === Math.ceil(filteredAds.length / advertsPerPage)
    : currentPage === Math.ceil(ads.length / advertsPerPage);

  // const handleFilterChange = event => {
  //   // //filterByName(event.target.value);
  //   // const filteredData = filterByName({ name: event.target.value, adverts });
  //   // setFilteredAdverts(filteredData);
  //   setFilteredAdverts(event.target.value);
  // };

  // const handleFilterChange = event => {
  //   const value = event.target.value;
  //   setFilteredAdverts(value);
  //   setIsFilterActive(value.trim() !== "");
  // };
  const handleFilterChange = event => {
    const value = event.target.value;
    setFilteredAdverts(value);
    setIsFilterActive(value.trim() !== "");
    dispatch(filterByName({ name: value.trim(), adverts: ads })); // Llama a la acci√≥n filterByName
  };

  return (
    <Layout title="anuncios">
      <>
        <section className="searchSection">
          <h1>Searching area</h1>
          <label className="advert_label">Name: </label>
          <input type="text" onChange={handleFilterChange} />
        </section>
        <div className="container">
          {isLoading ? (
            <div className="loadingPage">
              <div className="loadingInfo">
                <h1>LOADING....</h1>
              </div>
            </div>
          ) : (
            <div>
              {!!(ads && ads.length) ? (
                <>
                  <div className="listContainer">
                    <div className="contaienrTittle">
                      <h1>ADVERTISEMENTS AVIABLE</h1>
                    </div>
                    <ul>
                      {advertsToDisplay
                        .sort((a, b) => a.createdAt > b.createdAt)
                        .map(advert => (
                          <li key={advert.id}>
                            <div className="advert-container">
                              <Link to={`/adverts/${advert.id}`}>
                                <Advert
                                  advert={advert}
                                  onImageError={handleImageError}
                                />
                              </Link>
                            </div>
                          </li>
                        ))}
                    </ul>
                  </div>
                  <div className="pagination">
                    <p>
                      <span
                        className={currentPage === 1 ? "disabled" : "page"}
                        onClick={() => handlePageChange(currentPage - 1)}>
                        &lt;{" "}
                      </span>
                      {[...Array(totalPages)].map((_, index) => {
                        if (
                          totalPages > 5 &&
                          index > 1 &&
                          index < totalPages - 2
                        ) {
                          return (
                            <span className="page" key={index}>
                              ...
                            </span>
                          );
                        } else {
                          return (
                            <>
                              <span
                                className="page"
                                key={index}
                                onClick={() => handlePageChange(index + 1)}>
                                {index + 1}
                              </span>
                              {index < totalPages - 1 && <span> - </span>}
                            </>
                          );
                        }
                      })}
                      <span
                        className={isLastPage ? "disabled" : "page"}
                        onClick={() => handlePageChange(currentPage + 1)}>
                        {" "}
                        &gt;
                      </span>
                    </p>
                  </div>
                </>
              ) : (
                <EmptyList />
              )}
            </div>
          )}
        </div>
      </>
    </Layout>
  );
};

// const mapStateToProps = state => ({
//   adverts: getAdverts(state),
//   //isLoading: isLoading(state),
//   ...getUi(state),
// });
// const mapDispatchToProps = {
//   onAdvertsLoaded: advertsList,
// };

// export default connect(mapStateToProps, mapDispatchToProps)(AdvertsListPage);

export default AdvertsListPage;
