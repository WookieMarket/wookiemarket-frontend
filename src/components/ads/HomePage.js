import Layout from '../layout/Layout';
import { useTranslation } from 'react-i18next';
import AdvertsListPage from './AdvertsListPage';

const HomePage = (props) => {
  const { t } = useTranslation();
  return (
    <Layout title={t('welcome')} {...props}>
      <AdvertsListPage />
    </Layout>
  );
};

export default HomePage;
