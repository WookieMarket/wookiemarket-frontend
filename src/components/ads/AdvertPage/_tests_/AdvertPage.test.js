import AdvertPage from '../AdvertPage';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { createStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import ads from '../../../../store/slices/ads';
import auth from '../../../../store/slices/auth';
import ui from '../../../../store/slices/ui';

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key) => key,
    i18n: {
      changeLanguage: jest.fn(),
    },
  }),
}));

const rootReducer = combineReducers({
  ads: ads,
  auth: auth,
  ui: ui,
});

describe('AdvertPage', () => {
  const initialState = {
    ads: {
      areLoaded: true,
      data: [
        {
          id: '1',
          name: 'Advert 1',
          description: 'Advert 1 description',
          price: 100,
        },
        {
          id: '2',
          name: 'Advert 2',
          description: 'Advert 2 description',
          price: 200,
        },
      ],
    },
    ui: {
      isLoading: false,
      showModal: false,
      error: null,
    },
    auth: true,
  };
  const renderComponent = (advertId) => {
    const store = {
      getState: () => {
        return { ...initialState };
      },
      subscribe: () => {},
      dispatch: () => {},
    };
    return render(
      <Provider store={store}>
        <MemoryRouter initialEntries={[`/adverts/${advertId}`]}>
          <Routes>
            <Route path='/adverts/:advertId' element={<AdvertPage />} />
          </Routes>
        </MemoryRouter>
      </Provider>
    );
  };
  const renderAdvertPage = (initialEntries) => {
    const store = createStore(rootReducer, initialState);
    return render(
      <Provider store={store}>
        <MemoryRouter initialEntries={initialEntries}>
          <AdvertPage />
        </MemoryRouter>
      </Provider>
    );
  };

  test('Should handle navigation and useParams', () => {
    const advertId = '1';
    const advert = initialState.ads.data.find(
      (advert) => advert.id === advertId
    );

    renderComponent(advertId);

    expect(screen.getByText(advert.name)).toBeInTheDocument();
  });

  test('Should show advert data', () => {
    renderAdvertPage(['/adverts/1']);

    expect(screen.getByText('Advert 1')).toBeInTheDocument();
    expect(screen.getByText('Advert 1 description')).toBeInTheDocument();
    expect(screen.getByText('100')).toBeInTheDocument();
  });

  test('Should render correctly and show expected content', () => {
    renderAdvertPage(['/adverts/1']);

    expect(screen.getByText('ADVERT DETAIL')).toBeInTheDocument();
    expect(screen.getByText('Advert 1')).toBeInTheDocument();
    expect(screen.getByText('Advert 1 description')).toBeInTheDocument();
    expect(screen.getByText('100')).toBeInTheDocument();
  });
  test('Should match snapshot', () => {
    const { asFragment } = renderAdvertPage(['/adverts/1']);
    expect(asFragment()).toMatchSnapshot();
  });

  //Errors testing
  /*test('Should show error message when advert id does not exist', () => {
    const advertId = '999';
    renderComponent(advertId);

    expect(screen.getByText('Sorry, the requested ad is not available')).toBeInTheDocument();
  });*/
});
