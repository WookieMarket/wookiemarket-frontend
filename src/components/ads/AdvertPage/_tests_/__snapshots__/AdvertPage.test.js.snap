// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvertPage Should match snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <nav
      class="navbar"
    >
      <div
        class="icons"
      >
        <a
          href="/"
        >
          <h1
            class="title"
          >
             header 
          </h1>
        </a>
        <img
          alt="ES"
          class="icon-language"
          src="/images/languageIcons/es.png"
          title="icon-hover"
        />
        <img
          alt="EN"
          class="icon-language"
          src="/images/languageIcons/en.png"
          title="icon-hover"
        />
      </div>
      <ul>
        <li>
          <a
            class="navbar-list-item"
            href="/create-ad"
          >
            Upload ad
          </a>
          <a
            class="navbar-list-item"
            href="/ads"
          >
            Ads
          </a>
          <button
            class="navbar-list-item"
          >
            Logout
          </button>
        </li>
      </ul>
    </nav>
    <hr />
  </header>
  <div
    class="content"
  >
    <div
      class="holobackground"
    />
    <h1>
      ADVERT DETAIL
    </h1>
    <div
      class="productInfo"
      id="advertOnly"
    >
      <div
        class="advert-name"
      >
        <h2>
          Advert 1
        </h2>
      </div>
    </div>
    <br />
    <div
      class="productData"
    >
      <div
        class="product-img"
      >
        <img
          alt="imagenes anuncios"
          class="img"
        />
      </div>
      <br />
      <div
        class="product_Info"
      >
        <p
          class="advert_label"
        >
          Description: 
        </p>
        <span
          class="description"
        >
           Advert 1 description
        </span>
        <br />
        <br />
        <p>
          Is
          <span
            id="isSale"
          >
             purchased 
          </span>
          this product by:
        </p>
        <div
          class="price"
        >
          <p>
            100 
            <span
              id="price_coin"
            >
               â‚¬
            </span>
          </p>
        </div>
        <div
          class="advert_label"
        >
          <p>
            Category: 
            <span
              class="advert_text"
            />
          </p>
        </div>
      </div>
      <div>
        <p
          class="advert_label"
        >
          Username: 
          <span
            class="advert-text"
          />
        </p>
        <small
          class="advert_label"
        >
          Created at: 
          <span
            class="advert_text"
          >
            Invalid Date
          </span>
        </small>
      </div>
    </div>
    <div
      class="no-advert_content "
    />
  </div>
  <div
    class="holoBase "
  >
    Hola
  </div>
  <footer>
    <hr />
    <div
      class="footer"
    >
      <p>
        footer
      </p>
    </div>
  </footer>
</DocumentFragment>
`;
